sos/cleaner/mappings/ip_map.py

```diff
@@ -0,0 +1,189 @@
+        for net in self._networks:
+            if addr.ip == net.broadcast_address:
+                addr.network = net
+                return
+            if addr.ip in net:
+                nets.append(net)
+        # assign the address to the smallest network that was matched. This is
+        # necessary due to certain files specifying addresses that cause the
+        # ipaddress library to create artificially huge subnets that will
+        # include the actual subnets used by the system
+        if nets:
+            nets.sort(key=lambda n: n.prefixlen, reverse=True)
+            addr.network = nets[0]
+
+    def sanitize_item(self, item):
+        """Given an IP address, sanitize it to an obfuscated network or host
+        address as appropriate
+        """
+
+        addr = ipaddress.ip_interface(item)
+        network = addr.network
+
+        if str(network.netmask) == '255.255.255.255':
+            # check to see if this IP is in a subnet of an already obfuscated
+            # network and if it has, replace the default /32 netmask that
+            # ipaddress applies to no CIDR-notated addresses
+            self.set_ip_cidr_from_existing_subnet(addr)
+            return self.sanitize_ipaddr(addr)
+        else:
+            # we have a CIDR notation, so generate an obfuscated network
+            # address and then generate an IP address within that network's
+            # range
+            self.sanitize_network(network)
+            return self.sanitize_ipaddr(addr)
+
+    def sanitize_network(self, network):
+        """Obfuscate the network address provided, and if there are host bits
+        in the address then obfuscate those as well
+        """
+        # check if the address is in a network we've already encountered
+        if network not in self._networks:
+            self._new_obfuscated_network(network)
+
+    def sanitize_ipaddr(self, addr):
+        """Obfuscate the IP address within the known obfuscated network
+        """
+        # get the obfuscated network object
+        if addr.network in self._networks:
+            _obf_network = self._networks[addr.network]
+
+            # if the plain address is the broadcast address for it's own
+            # network, then assign the broadcast address for the obfuscated
+            # network
+            if addr.ip == addr.network.broadcast_address:
+                return str(_obf_network.broadcast_address)
+
+            # otherwise within that obfuscated network grab the next available
+            # address from it
+            for _ip in _obf_network.hosts():
+                if not self.ip_in_dataset(_ip):
+                    # the ipaddress module does not assign the network's
+                    # netmask to hosts in the hosts() generator for some reason
+                    return "%s/%s" % (str(_ip), _obf_network.prefixlen)
+
+        # ip is a single ip address without the netmask
+        return self._new_obfuscated_single_address()
```

Comment:
Can't we generate the same obfuscated IP address for different input addresses, here?

Well, the probability is small, as we are picking from a range of 65M (90^4) addresses. But if my math is correct, then e.g. there is 50% chance of hitting already picked address when obfuscating 9538 different IP addresses (or 10% chance for 3719 addresses).

Reply:
Easy enough to add a check here, but yeah my thought initially was that it was an incredibly probability.